### PR TITLE
#611 - Allowed unauthorized ProtocolsQuery to fetch published ProtocolsConfigure

### PR DESCRIPTION
**Pull request by thehenrytsai**

### CODE REVIEW

File: tests/handlers/protocols-query.spec.ts [hunk 1]
```diff
@@ -117,23 +120,45 @@ export function testProtocolsQueryHandler(): void {
...
         const conditionalQueryReply = await dwn.processMessage(alice.did, conditionalQuery.message);
-
         expect(conditionalQueryReply.status.code).to.equal(200);
         expect(conditionalQueryReply.entries?.length).to.equal(1); // only 1 entry should match the query on protocol
 
         const protocolConfigured = conditionalQueryReply.entries![0] as ProtocolsConfigureMessage;
         expect(protocolConfigured).to.deep.equal(protocol2.message);
 
-        // testing fetch-all query without filter
+        // testing authenticated but unauthorized (to private ProtocolsConfigures) conditional query
```
Comment (LiranCohen) [inline tests/handlers/protocols-query.spec.ts:129]: The "to private ProtocolsConfigures" part of the comment was confusing a bit for me, had to re-read it a couple of times before I understood what it meant.
Maybe something like this is clearer?

```suggestion
        // testing authenticated but unauthorized conditional query. It should return only public messages.
```

File: tests/handlers/protocols-query.spec.ts [hunk 2]
```diff
@@ -117,23 +120,45 @@ export function testProtocolsQueryHandler(): void {
...
         });
 
         const fetchAllQueryReply = await dwn.processMessage(alice.did, fetchAllQuery.message);
-
         expect(fetchAllQueryReply.status.code).to.equal(200);
         expect(fetchAllQueryReply.entries?.length).to.equal(2);
         expect(fetchAllQueryReply.entries).to.deep.include(protocol2.message);
         expect(fetchAllQueryReply.entries).to.deep.include(protocol3.message);
+
+        // testing authenticated but authorized (to private ProtocolsConfigures) fetch-all query without filter
```
Comment (LiranCohen) [inline tests/handlers/protocols-query.spec.ts:152]: Same comment as above, but also  thing you meant unauthorized here.
```suggestion
        // testing authenticated but unauthorized (to private ProtocolsConfigures) fetch-all query without filter
```

**Review by LiranCohen — COMMENTED** [review]
Looks good! Just a couple of nits wrt the comments in the testing section.

Also the PR should be renamed to `ProtocolsQuery` instead of `RecordsQuery`.

File: tests/handlers/protocols-query.spec.ts (hunk 1 above)
Comment (thehenrytsai) [reply tests/handlers/protocols-query.spec.ts:129]: Fixed!

File: tests/handlers/protocols-query.spec.ts (hunk 2 above)
Comment (thehenrytsai) [reply tests/handlers/protocols-query.spec.ts:152]: Fixed!

**Review by LiranCohen — APPROVED** [review]

**Pull request merged**